plugins/outputs/parquet/parquet.go

```diff
@@ -0,0 +1,339 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package parquet
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/apache/arrow/go/v18/arrow"
+	"github.com/apache/arrow/go/v18/arrow/array"
+	"github.com/apache/arrow/go/v18/arrow/memory"
+	"github.com/apache/arrow/go/v18/parquet"
+	"github.com/apache/arrow/go/v18/parquet/pqarrow"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type metricGroup struct {
+	filename string
+	builder  *array.RecordBuilder
+	schema   *arrow.Schema
+	writer   *pqarrow.FileWriter
+}
+
+type Parquet struct {
+	Directory          string          `toml:"directory"`
+	RotationInterval   config.Duration `toml:"rotation_interval"`
+	TimestampFieldName *string         `toml:"timestamp_field_name"`
+	Log                telegraf.Logger `toml:"-"`
+
+	metricGroups map[string]*metricGroup
+}
+
+func (*Parquet) SampleConfig() string {
+	return sampleConfig
+}
+
+func (p *Parquet) Init() error {
+	if p.Directory == "" {
+		p.Directory = "."
+	}
+
+	if p.TimestampFieldName == nil {
+		timestampFieldName := "timestamp"
+		p.TimestampFieldName = &timestampFieldName
+	}
```

Comment:
Alternatively you could use a non-nil struct member and set the default in the `init()` (lowercase) function allowing an explicit overwrite with an empty string...

Reply:
The adaptation is not yet in the code... Did you maybe forget to push that commit?

Reply:
Are you looking at the "outdated" suggestion, as this whole code block is removed ;)

Reply:
Hmmm I don't think so... In the `parquet.go` file line 37 I see
```go
	TimestampFieldName *string         `toml:"timestamp_field_name"`
```

so the `TimestampFieldName` is still a pointer isn't it?

Reply:
Pushed update to change it to a string, non-pointer